pkg/sql/catalog/internal/catkv/system_database_cache.go

```diff
@@ -75,11 +77,15 @@ func NewSystemDatabaseCache(codec keys.SQLCodec, settings *cluster.Settings) *Sy
 func (c *SystemDatabaseCache) lookupDescriptor(
 	_ clusterversion.ClusterVersion, id descpb.ID,
 ) catalog.Descriptor {
-	if id == keys.SystemDatabaseID {
+	switch id {
+	case keys.SystemDatabaseID:
 		return systemschema.SystemDB
+	case keys.SystemPublicSchemaID:
+		return schemadesc.GetPublicSchema()
```

Comment:
The public schema of the system database is unfortunately still very descriptorless so it made sense to encode that fact here.